rules/framework/vfs_overlay.bzl

```diff
@@ -69,15 +69,17 @@ def _build_subtrees(paths, vfs_prefix):
                 break
 
             # Lookup a value for the current subdirs, otherwise append
-            next_subdirs = curr_subdirs.dict.get(part, None)
+            if curr_subdirs:
+                next_subdirs = curr_subdirs.dict.get(part, None)
+            else:
+                next_subdirs = None
             if not next_subdirs:
                 next_subdirs_json = {"contents": [], "type": "directory", "name": part}
                 next_subdirs = struct(dict = {}, json = next_subdirs_json)
 
                 curr_subdirs.dict[part] = next_subdirs
                 curr_subdirs.json["contents"].append(next_subdirs_json)
-
-            curr_subdirs = next_subdirs
+                curr_subdirs = next_subdirs
```

Comment:
@tinder-maxwellelliott was this indentation intentional? Getting a failure in our internal builds and if I clean => undo this indentation => build, it goes 🟢, if I revert and repeat it goes 🔴 

Still couldn't repro in a `rules_ios` target but regarding wanted to check if this was intentional 🤔

Reply:
I had to move this to get things working. I think we need to add a test to ensure this safeguards your internal builds

Reply:
Right thx for checking!

Reply:
@tinder-maxwellelliott would you be able to share more details about the error you're hitting without this change? I don't want to revert and break things for others so I'll have to find a fix that applies for both scenarios, then we can add tests to safeguard both use cases

Reply:
Yeah that specific change is the existing code updated `current_subdirs` to always be `next_subdirs` on each click of the loop. This code has had some tricky edge cases to get right to say the least 🙃 

> I think we need to add a test to ensure this safeguards

Yeah we should have more permutations tested here and also this might have this fixed existing cases that were disabled on CI!